api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,15 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
+				&& ((this.getEndDate() != null && o.getEndDate() != null
+				&& this.getEndDate().equals(o.getEndDate()))
+				|| (this.getEndDate() == null && o.getEndDate() == null))) {
+			return 0;
+		} else {
+			return -1;
+		}
```

Comment:
I guess my question should have been why you chose return -1 over 1, am not suggesting another over the other, implementations of compareTo typically return 1, 0 or -1 but I see you only return just two of them. You could  chose to return -1 if startDate is earlier otherwise 1 for active memberships. And always push stopped memberships to the end of the collections with the one with the latest start date coming dead last.